src/lib/containers/UpsetPlot.tsx

```diff
@@ -48,65 +50,72 @@ const UpsetPlot: React.FunctionComponent<UpsetPlotProps> = ({
-        entityId,
-        query: {
-          sql
-        },
-      }
-      const {
-        queryResult
-      } = await SynapseClient.getFullQueryTableResults(
-        queryRequest,
-        token
-      )
-      // transform query data into plot data, and store.
-      // collect all values for each key
-      const keyValuesMap = {}
-      // keyValuesMap looks like { 'A': {name: 'A', sets: ['S1', 'S2'] }, 'B': { name: 'B', sets: ['S1'] }, ... }
-      // It's a little redudant, but makes the next step much easier.
-      
-      const caseInsensitiveSetNames = {}
-      // caseInsensitiveSetNames looks like { 'RNASEQ': 'rnaSeq', 'NOMe-SEQ': 'NOMe-seq'}.
+      try {
+        const entityId = parseEntityIdFromSqlStatement(sql)
+        const partMask = SynapseConstants.BUNDLE_MASK_QUERY_RESULTS
+        const queryRequest: QueryBundleRequest = {
+          partMask,
+          concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+          entityId,
+          query: {
+            sql
+          },
+        }
+        const {
+          queryResult
+        } = await SynapseClient.getFullQueryTableResults(
+          queryRequest,
+          token
+        )
+        // transform query data into plot data, and store.
+        // collect all values for each key
+        const keyValuesMap = {}
+        // keyValuesMap looks like { 'A': {name: 'A', sets: ['S1', 'S2'] }, 'B': { name: 'B', sets: ['S1'] }, ... }
+        // It's a little redudant, but makes the next step much easier.
+        
+        const caseInsensitiveSetNames = {}
+        // caseInsensitiveSetNames looks like { 'RNASEQ': 'rnaSeq', 'NOMe-SEQ': 'NOMe-seq'}.
 
-      for (const row of queryResult.queryResults.rows) {
-        for (let j = 1; j < row.values.length; j += 1) {
-          const rowValues:string[] = row.values
-          const key = rowValues[0]
-          let newValue = rowValues[j]
-          keyValuesMap[key] = keyValuesMap[key] || {}
-          keyValuesMap[key].name = key
+        for (const row of queryResult.queryResults.rows) {
+          for (let j = 1; j < row.values.length; j += 1) {
+            const rowValues:string[] = row.values
+            const key = rowValues[0]
+            let newValue = rowValues[j]
+            keyValuesMap[key] = keyValuesMap[key] || {}
+            keyValuesMap[key].name = key
 
-          keyValuesMap[key].sets = keyValuesMap[key].sets || []
-          const found = keyValuesMap[key].sets.find((v: any) => v == newValue)
-          if (!found) {
-            if (caseInsensitiveSetNames[newValue.toUpperCase()]) {
-              newValue = caseInsensitiveSetNames[newValue.toUpperCase()]
-            } else {
-              caseInsensitiveSetNames[newValue.toUpperCase()] = newValue
+            keyValuesMap[key].sets = keyValuesMap[key].sets || []
+            const found = keyValuesMap[key].sets.find((v: any) => v == newValue)
+            if (!found && newValue) {
```

Comment:
This is the bug fix (to verify newValue is defined before pushing it into the set)